e2e/vault/README.md

```diff
@@ -0,0 +1,12 @@
+# Vault Integration Test
+
+Downloads, caches, and tests Nomad against open source Vault binaries. Runs
+only when `NOMAD_E2E` is set.
+
+Run with:
+
+```
+NOMAD_E2E=1 go test
```

Comment:
I'd update this to specifically call out running this test, i.e. `NOMAD_E2E=1 go test -run TestVaultCompatibility`

If we implement this in the e2e Framework, we could mark it as a “slow” test when we running the tests locally against, say, AWS clusters. but until the stuff I’m working on for provisioning updates lands (tomorrowish?) those constraints don’t really seem to work completely so probably better to keep it as-is.